aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/RealAWSCognitoAuthPlugin.kt

```diff
@@ -2223,45 +2224,57 @@ internal class RealAWSCognitoAuthPlugin(
         onSuccess: Action,
         onError: Consumer<AuthException>
     ) {
-        authStateMachine.getCurrentState { authState ->
-            when (authState.authNState) {
-                is AuthenticationState.SignedIn -> {
-                    GlobalScope.launch {
-                        try {
-                            val accessToken = getSession().userPoolTokensResult.value?.accessToken
-                            accessToken?.let { token ->
-                                authEnvironment.cognitoAuthService.cognitoIdentityProviderClient?.setUserMfaPreference {
-                                    this.accessToken = token
-                                    this.smsMfaSettings = sms?.let {
-                                        SmsMfaSettingsType.invoke {
-                                            enabled = it.mfaEnabled
-                                            it.mfaPreferred ?.let { preferred -> preferredMfa = preferred }
-                                        }
-                                    }
-                                    this.softwareTokenMfaSettings = totp?.let {
-                                        SoftwareTokenMfaSettingsType.invoke {
-                                            enabled = it.mfaEnabled
-                                            it.mfaPreferred ?.let { preferred -> preferredMfa = preferred }
+        if (sms == null && totp == null) {
+            onError.accept(InvalidParameterException())
```

Comment:
A few issues here.

If we are going to callback in onError we need to return so we don't continue with another callback.

Also, the default messaging for InvalidParameterException is not very helpful. What does iOS say in this instance?

Reply:
This is a good change. Lets take the extra step to add in, but improve the default message/recovery suggestion to help the customer here.

Reply:
done.